tests/unit/UserValidatorTest.php

```diff
@@ -0,0 +1,198 @@
+
+        $validator = $this->validator
+            ->bind('name', ['user_id' => 1])
+            ->bind('email', ['user_id' => 1]);
+
+        $data = [
+            'email'         => 'test@example.com',
+            'name'          => 'John Doe',
+            'language_code' => 'xx'
+        ];
+
+        try {
+            $validator->validate('updateMe', $data);
+        } catch (ValidationException $exception) {
+            $this->assertNotSame(
+                false,
+                array_search(trans('gzero-core::user.invalid_language'), $exception->errors()['language_code']));
+            return;
+        }
+
+        $this->fail('It should throw an exception');
+    }
+
+    /** @test */
+    public function canValidateInactiveLanguageCode()
+    {
+        $this->tester->getApplication()->instance(LanguageService::class, new LanguageService(
+            collect([
+                new Language(['code' => 'en', 'is_enabled' => true, 'is_default' => true]),
+                new Language(['code' => 'pl', 'is_enabled' => false, 'is_default' => false]),
+            ])
+        ));
+
+        $validator = $this->validator
+            ->bind('name', ['user_id' => 1])
+            ->bind('email', ['user_id' => 1]);
+
+        $data = [
+            'email'         => 'test@example.com',
+            'name'          => 'John Doe',
+            'language_code' => 'pl'
+        ];
+
+        try {
+            $validator->validate('updateMe', $data);
+        } catch (ValidationException $exception) {
+            $this->assertNotSame(
+                false,
+                array_search(trans('gzero-core::user.invalid_language'), $exception->errors()['language_code']));
+            return;
+        }
+
+        $this->fail('It should throw an exception');
+    }
+
+    /** @test */
+    public function canValidateActiveLanguageCode()
+    {
+        $this->tester->getApplication()->instance(LanguageService::class, new LanguageService(
+            collect([
+                new Language(['code' => 'en', 'is_enabled' => true, 'is_default' => true]),
+                new Language(['code' => 'pl', 'is_enabled' => false, 'is_default' => false]),
+            ])
+        ));
+
+        $validator = $this->validator
+            ->bind('name', ['user_id' => 1])
+            ->bind('email', ['user_id' => 1]);
+
+        $data = [
+            'email'         => 'test@example.com',
+            'name'          => 'John Doe',
+            'language_code' => 'en'
+        ];
+
+        $validator->validate('updateMe', $data);
+    }
+
+    /** @test */
+    public function canValidateWithoutLanguageCode()
+    {
+        $this->validator
+            ->bind('name', ['user_id' => 1])
+            ->bind('email', ['user_id' => 1])
+            ->validate('updateMe', [
+                'email' => 'test@example.com',
+                'name'  => 'John Doe',
+            ]);
+    }
+
+    /** @test */
+    public function canValidateWithoutTimezone()
```

Comment:
this test is exactly the same as `canValidateWithoutLanguageCode` and `canValidateUsingRulesInMethod` there are no assertions in them so it's hard to figure out what is the difference - if any.